pkg/operator/sync.go

```diff
@@ -62,3 +69,51 @@ func (c *csiSnapshotOperator) waitForCustomResourceDefinition(resource *apiextv1
 	}
 	return nil
 }
+
+func (c *csiSnapshotOperator) syncDeployments() error {
+	deploy := resourceread.ReadDeploymentV1OrDie(generated.MustAsset(deployment))
+	_, updated, err := resourceapply.ApplyDeployment(c.deployClient, c.eventRecorder, deploy, 0, true)
+	if err != nil {
+		return err
+	}
+	if updated {
+		if err := c.waitForDeploymentRollout(deploy); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+//nolint:dupl
+func (c *csiSnapshotOperator) waitForDeploymentRollout(resource *appsv1.Deployment) error {
```

Comment:
Is this method copied from somewhere? It looks odd to wait for `UnavailableReplicas==0`. We probably want to set `ClusterOperator.Status.Conditions["Progressing"]=true` and then process events on Deployment and update the conditions as the Deployment is rolled out.